tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AspNetWebApi2Integration.cs

```diff
@@ -62,12 +65,27 @@ internal static Scope CreateScope(IHttpControllerContext controllerContext, out
                     {
                         Log.Error(ex, "Error extracting propagated HTTP headers.");
                     }
+
+                    if (!tracer.Settings.IpHeaderDisabled)
+                    {
+                        if (request.Properties.TryGetValue("MS_OwinContext", out var owinContextObj))
```

Comment:
This looks good for the OWIN scenario! I think we should also do the same when the WebApi is hosted in IIS / System.Web.dll. To do that, it should be very similar to this, with the following differences ([see this reference](https://stackoverflow.com/a/19849122)):
- The value can be extracted by getting `request.Properties["MS_HttpContext"]`
- The peerIpAddress value can be extracted from `request.Properties["MS_HttpContext"].IsSecureConnection`
- The isSecureConnection value can be extracted from `request.Properties["MS_HttpContext"].UserHostAddress`

The StackOverflow link also has a "self-hosted" scenario but I don't really know anything about that scenario, so I'm not sure if it's necessary :shrug:

Reply:
> The StackOverflow link also has a "self-hosted" scenario but I don't really know anything about that scenario, so I'm not sure if it's necessary 🤷

This was basically a way of creating "web apis" using the WCF libraries _System.ServiceModel_ ([This SO post describes it a bit](https://stackoverflow.com/a/44568711)). I'm sad to say I've actually done this in the past, but this is _long_ deprecated, and I don't think there's any reason to support it specifically...